tapestry.js

```diff
@@ -30,11 +31,11 @@ const // calculated
 var dataset, root, svg, links, nodes,               // Basics
     path, pieGenerator, arcGenerator,               // Donut
     linkForce, collideForce, force,                 // Force
-    tapestrySlug, saveProgress = true,      // Cookie
+    tapestrySlug, saveProgress = true, saveProgressCounter = 0,     // Cookie
```

Comment:
Please change `saveProgressCounter` to be a timestamp instead of an integer and call it `progressLastSaved`. The reason is that this way, we are not relying on the `ontimeupdate` event to count our seconds (it may happen at a shorter or longer interval than 1s). So your code here should be something like `progressLastSaved = Date()`.